sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -56,6 +56,16 @@ class CodeGenContext {
    */
   val references: mutable.ArrayBuffer[Expression] = new mutable.ArrayBuffer[Expression]()
 
+  val mutableStates: mutable.ArrayBuffer[String] = mutable.ArrayBuffer.empty[String]
```

Comment:
ISTM that all the kinds of generated stuffs should be put in GeneratedExpressionCode, not in CodeGenContext
according to the source code comments.